### PR TITLE
PE-785 - Adding custom course home feature

### PR DESCRIPTION
**Pull request by diegomillan**
This PR supports POLS integration

### CODE REVIEW

File: lms/djangoapps/courseware/views/views.py [hunk 1]
```diff
@@ -312,6 +313,14 @@ def course_info(request, course_id):
 
     Assumes the course_id is in a valid format.
     """
+    custom_course_home = run_extension_point(
+        'OEE_COURSE_HOME_CALCULATOR',
+        course_id=course_id,
+        user=request.user,
+    )
```
Comment (Squirrel18) [inline lms/djangoapps/courseware/views/views.py:320]: Please, add a break line here. @diegomillan

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+from openedx.core.djangoapps.site_configuration import helpers as configuration_helpers
+
+log = logging.getLogger(__name__)
+
+
+def run_extension_point(extension_point, *args, **kwargs):
+    """
+    Wrapper function to execute any extension point at platform level
+    If exceptions occurs returns None
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py]: Period at the end of the line. @diegomillan

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    """
+    path = None
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py:19]: Please, @diegomillan add a break line here.

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    try:
+        path = configuration_helpers.get_value(
+            extension_point,
+            getattr(settings, extension_point, None)
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py]: Trailing comma. @diegomillan

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+        )
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py:25]: Please, add a break line here. @diegomillan

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        if not path:
+            return
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py]:
```suggestion
            return None
```

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    except AttributeError:
+        return
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py]:
```suggestion
        return None
```

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+    try:
+        module, func_name = path.rsplit('.', 1)
+        module = import_module(module)
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py]: module_name, func_name = path.rsplit('.', 1)
module = import_module(module_name)

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+        extension_function = getattr(module, func_name)
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py:35]: Please add a default value for getattr.
Please add a break line here. @diegomillan

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+        return extension_function(*args, **kwargs)
+    except ImportError:
+        log.info('Could not import the %s : %s', extension_point, module)
+        return
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py]:
```suggestion
        return None
```

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+    except AttributeError:
+        log.info('Could not import the function %s in the module %s', func_name, module)
+        return
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py]:
```suggestion
        return None
```

File: openedx/core/djangoapps/plugins/plugin_extension_points.py [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+    except ValueError:
+        log.info('Could not load the information from \"%s\"', path)
+        return
```
Comment (Squirrel18) [inline openedx/core/djangoapps/plugins/plugin_extension_points.py]:
```suggestion
        return None
```

File: openedx/features/course_experience/views/course_home.py [hunk 13]
```diff
@@ -58,6 +60,14 @@ def get(self, request, course_id, **kwargs):
         """
         Displays the home page for the specified course.
         """
+        custom_course_home = run_extension_point(
+            'OEE_COURSE_HOME_CALCULATOR',
+            course_id=course_id,
+            user=request.user,
+        )
```
Comment (Squirrel18) [inline openedx/features/course_experience/views/course_home.py:67]: Please, add a break line here. @diegomillan

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

File: openedx/core/djangoapps/plugins/plugin_extension_points.py (hunk 9 above)
Comment (diegomillan) [reply openedx/core/djangoapps/plugins/plugin_extension_points.py:35]: > Please add a default value for getattr.
> Please add a break line here. @diegomillan

In this case, the default attr was not set intentionally.  AttributeError handles the case when an error occurs.

Comment (Squirrel18) [reply openedx/core/djangoapps/plugins/plugin_extension_points.py:35]: Got it. @diegomillan

File: openedx/core/djangoapps/plugins/plugin_extension_points.py (hunk 8 above)
Comment (Squirrel18) [reply openedx/core/djangoapps/plugins/plugin_extension_points.py]: Please @diegomillan Address this one.

**Review by Squirrel18 — APPROVED** [review]

**Pull request merged**